src/Utilities/StaticCache.hpp

```diff
@@ -0,0 +1,120 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <utility>
+
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace StaticCache_detail {
+template <typename T, typename... Ranges>
+class StaticCacheImpl;
+
+template <typename T, typename Range0, typename... Ranges>
+class StaticCacheImpl<T, Range0, Ranges...> {
+  template <typename>
```

Comment:
see if you can get rid of this by using a default template parameter of index_sequence